iradix.go

```diff
@@ -183,6 +183,32 @@ func (t *Txn) writeNode(n *Node, forLeafUpdate bool) *Node {
 	return nc
 }
 
+// Visit all the nodes in the tree under n, and add their mutateChannels to the transaction
+func (t *Txn) recursiveWatchNotify(n *Node, fn func(n *Node) bool) bool {
+
+	// Mark this node as being mutated.
+	if t.trackMutate {
+		t.trackChannel(n.mutateCh)
+	}
+
+	// Mark its leaf as being mutated, if appropriate.
+	if t.trackMutate && n.leaf != nil {
+		t.trackChannel(n.leaf.mutateCh)
+	}
+
+	if n.leaf != nil && fn(n) {
```

Comment:
Why do we check the callback's return value here? That always returns `false` so this counts as a side effect (for the leaf) but exits in kind of an obscure way. I think we should get rid of the callback and just have this keep track of the count and return it; that'll simplify things and make the behavior super clear, since we don't have any other callback behavior to hook into here.